indexer/ai/src/main/java/com/google/cloudsearch/ai/AISkillBase.java

```diff
@@ -0,0 +1,123 @@
+package com.google.cloudsearch.ai;
+
+import com.google.cloudsearch.exceptions.InvalidConfigException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.util.*;
+
+/**
+ * Abstract class AISkillBase implements the common functionality required for the different skills.
+ */
+public abstract class AISkillBase implements AISkill {
+
+    private String aiSkillName = "";
+    private Map<String, List<OutputMapping>> outputMappings = new HashMap<>();
+    private Map<String, Boolean> schemaInfo = new HashMap<String, Boolean>();
+    private Boolean isSchemaInfoStored = false;
+
+    /**
+     * Parses the CloudsSearch Schema and stores the object and property names
+     * @param schema    CloudSearch schema
+     */
+    private void storeSchemaInfo(JSONObject schema){
+        JSONArray schemaObjects = (JSONArray) schema.get("objectDefinitions");
+        for(Object schemaObj : schemaObjects){
+            JSONObject obj = (JSONObject) schemaObj;
+            String objName = (String) obj.get("name");
+            JSONArray propertyDefinitions = (JSONArray) obj.get("propertyDefinitions");
+            for(Object propertyObj : propertyDefinitions){
+                JSONObject property = (JSONObject) propertyObj;
+                String propertyName = (String) property.get("name");
+                schemaInfo.put(objName+"."+propertyName, true);
+            }
+        }
+        this.isSchemaInfoStored = true;
+    }
+
+    /**
+     * Helper function to validate property names in config
+     * @param propertyName  object name + property name specified in configuration
+     * @param schema    CloudSearch schema
+     * @return
+     */
+    private boolean isValid(String propertyName, JSONObject schema){
+
+        if(isSchemaInfoStored == false)
```

Comment:
redundant new line.

Reply:
function name is not clear whether whats its role

Reply:
done